modules/generative-anthropic/config/class_settings.go

```diff
@@ -0,0 +1,160 @@
+	return &classSettings{cfg: cfg, propertyValuesHelper: basesettings.NewPropertyValuesHelper("generative-anthropic")}
+}
+
+func (ic *classSettings) Validate(class *models.Class) error {
+	if ic.cfg == nil {
+		// we would receive a nil-config on cross-class requests, such as Explore{}
+		return errors.New("empty config")
+	}
+	model := ic.getStringProperty(modelProperty, DefaultAnthropicModel)
+
+	if model == nil || !ic.validateModel(*model) {
+		return errors.Errorf("wrong Anthropic model name, available model names are: %v", availableAnthropicModels)
+	}
+
+	return nil
+}
+
+func (ic *classSettings) getStringProperty(property string, defaultValue string) *string {
+	asString := ic.propertyValuesHelper.GetPropertyAsStringWithNotExists(ic.cfg, property, "", defaultValue)
+	return &asString
+}
+
+func (ic *classSettings) getIntProperty(name string, defaultValue *int) *int {
+	var wrongVal int = -1
+	return ic.propertyValuesHelper.GetPropertyAsIntWithNotExists(ic.cfg, name, &wrongVal, defaultValue)
+}
+
+func (ic *classSettings) getFloatProperty(name string, defaultValue *float64) *float64 {
+	var wrongVal float64 = -1.0
+	return ic.propertyValuesHelper.GetPropertyAsFloat64WithNotExists(ic.cfg, name, &wrongVal, defaultValue)
+}
+
+func (ic *classSettings) getListOfStringsProperty(name string, defaultValue []string) *[]string {
+	if ic.cfg == nil {
+		// we would receive a nil-config on cross-class requests, such as Explore{}
+		return &defaultValue
+	}
+
+	model, ok := ic.cfg.ClassByModuleName("generative-anthropic")[name]
+	if ok {
+		asStringList, ok := model.([]string)
+		if ok {
+			return &asStringList
+		}
+		var empty []string
+		return &empty
+	}
+	return &defaultValue
+}
+
+func (ic *classSettings) GetMaxTokensForModel(model string) int {
+	return defaultMaxTokens[model]
+}
+
+// possibly we just validate the model name is not empty, to allow
+// for future models to be added without changing the code
+func (ic *classSettings) validateModel(model string) bool {
+	return contains(availableAnthropicModels, model)
+}
+
+func (ic *classSettings) BaseURL() string {
+	return *ic.getStringProperty(baseURLProperty, DefaultBaseURL)
+}
+
+func (ic *classSettings) Model() string {
+	return *ic.getStringProperty(modelProperty, DefaultAnthropicModel)
+}
+
+func (ic *classSettings) MaxTokens() int {
+	return *ic.getIntProperty(maxTokensProperty, &DefaultAnthropicMaxTokens)
+}
+
+func (ic *classSettings) Temperature() float64 {
+	return *ic.getFloatProperty(temperatureProperty, &DefaultAnthropicTemperature)
+}
+
+func (ic *classSettings) K() int {
+	return *ic.getIntProperty(topKProperty, &DefaultAnthropicK)
+}
+
+func (ic *classSettings) P() float64 {
+	return *ic.getFloatProperty(topPProperty, &DefaultAnthropicP)
+}
+
+func (ic *classSettings) StopSequences() []string {
+	return *ic.getListOfStringsProperty(stopSequencesProperty, DefaultAnthropicStopSequences)
+}
+
+func contains[T comparable](s []T, e T) bool {
```

Comment:
No need to declare this method, you can use `basesettings.ValidateSetting(...)` method, it does the same